runtime/vm/segment.c

```diff
@@ -543,7 +543,7 @@ allLiveClassesNextDo(J9ClassWalkState* state)
 	J9Class* clazzPtr = NULL;
 	J9Class* clazzPtrCandidate = NULL;
 	J9JavaVM* vm = state->vm;
-	const BOOLEAN needCheckInGC = (J9_GC_WRITE_BARRIER_TYPE_REALTIME == vm->gcWriteBarrierType);
+	const BOOLEAN needCheckInGC = ((J9_GC_WRITE_BARRIER_TYPE_SATB == vm->gcWriteBarrierType) || (J9_GC_WRITE_BARRIER_TYPE_SATB_AND_OLDCHECK == vm->gcWriteBarrierType));
```

Comment:
I believe this really meant Metronome GC policy (specifically a GC policy that will do incremental or concurrent unloading). Since Standard GC with SATB will still be doing unloading in one shot, this extra liveness check is likely not needed there.

Reply:
Sounds good, it can be added in later if needed. Since `J9_GC_WRITE_BARRIER_TYPE_REALTIME` barrier type is changed to `J9_GC_WRITE_BARRIER_TYPE_SATB ` and shared, I will change the barrier check to policy check.

Reply:
I took a deeper look into what checkClassLive does. And while it is concerned about aspect of incremental root clearing and possibly unloading (path that when _unmarkedImpliesClasses is true) that are Metronome specific, it also has an action even if marking is in a middle of cycle (_unmarkedImpliesClasses is false) what would be valid case for Standard SATB too.

Class marking/tracing is different in Metronome and Standard collectors, and to be honest I don't know if that _else_ close is really necessary or not for Standard SATB.

To be on a safe side, it would be wise to restore it to your original variant that is more generic.